src/main/groovy/com/twcable/grabbit/jcr/JCRNodeDecorator.groovy

```diff
@@ -105,6 +106,37 @@ class JCRNodeDecorator {
         return definition.isMandatory()
     }
 
+    void checkoutIfNecessary(javax.jcr.Node node) throws RepositoryException {
+        try{
+            javax.jcr.Node versionableNode = findVersionableAncestor(node);
+            if (versionableNode != null) {
+                if (!versionableNode.isCheckedOut()) {
+                    versionableNode.checkout();
+                }
+            }
+        }
+        catch (Exception exception){
+            log.warn "Could not checkout node ${node.path}"
+        }
+
+    }
+
+    private javax.jcr.Node findVersionableAncestor(javax.jcr.Node node) throws RepositoryException {
```

Comment:
Not sure I understand why we are doing this. For simplicities sake, I think we should only care about writing to this particular node we are wrapping.